internal/auth/legacy.go

```diff
@@ -0,0 +1,110 @@
+package auth
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"sync"
+
+	"golang.org/x/oauth2"
+
+	"github.com/platformsh/cli/internal/legacy"
+)
+
+type legacyCLITokenSource struct {
+	ctx     context.Context
+	cached  *oauth2.Token
+	wrapper *legacy.CLIWrapper
+	mu      sync.Mutex
+}
+
+func (ts *legacyCLITokenSource) unsafeGetLegacyCLIToken() (*oauth2.Token, error) {
+	bt := bytes.NewBuffer(nil)
+	ts.wrapper.Stdout = bt
```

Comment:
minor - If two parts of the same `cli` call the legacy-cli concurrently, this `ts.wrapper.Stdout` assignment could race; perhaps it should be a function argument.